src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/IDependencyContextProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.DependencyModel;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationParts
+{
+    /// <summary>
+    /// Exposes a <see cref="DependencyContext"/> from an <see cref="ApplicationPart"/>.
+    /// </summary>
+    public interface IDependencyContextProvider
```

Comment:
I believe the interesting part of CompileLibraries is https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.Extensions.DependencyModel/CompilationLibrary.cs#L39 that we use on https://github.com/aspnet/Mvc/pull/4672/files#diff-ab147a7292f31e46ebc6b39f1df7480bR49

Maybe we can have our own DTO with the interesting parts of compilelibrary and the resolved paths? I think that strikes a good balance. I think something like ApplicationReference { Name, Version and Paths } will be enough

@rynowak @pranavkm Thoughts?